File: Day2/MaxAreaOfIslands.java
package Day2;

public class MaxAreaOfIslands {
   public int max=0;
    public int maxAreaOfIsland(int[][] grid) {
        int ret=0;
        for (int i = 0; i < grid.length; i++) {
            for (int j = 0; j < grid.length; j++) {
                if(grid[i][j]==1)
                {
                    max=0;
                    dfs(i, j, grid);
                }
                ret=Math.max(max, ret);
            }
        }
        return ret;
    }

    public void dfs(int i,int j,int [][]grid)
    {
        if(i>=0&&j>=0&&i<grid.length&&j<grid[0].length&&grid[i][j]==1)
        {
            grid[i][j]=2;
            max++;
           
            dfs(i-1, j, grid);
            dfs(i+1, j, grid);
            dfs(i, j-1, grid);
            dfs(i, j+1, grid);
            

        }

    }
}

File: Day2/MaxAreaOfIslands.py
from typing import List


class Solution(object):
    global ma
    def sol(self,grid:List[List[str]])->int:
        ret=0
        for i in range(len(grid)):
            for j in range(len(grid[0])):
                if grid[i][j]=='1':
                    ma=0
                    self.dfs(grid,i,j)
                    ret=max(ret,ma)
        return ret            
    
    def dfs(self,grid,i,j):
        if i<0 or j<0 or i>=len(grid) or j>=len(grid[0]) or grid[i][j]!='1':
            return
        grid[i][j]='+'
        ma+=1
        self.dfs(grid,i+1,j)
        self.dfs(grid,i-1,j)
        self.dfs(grid,i,j-1)
        self.dfs(grid,i,j+1)


